stan/math/fwd/functor/fvar_finite_diff.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef STAN_MATH_FWD_FUNCTOR_FVAR_FINITE_DIFF_HPP
+#define STAN_MATH_FWD_FUNCTOR_FVAR_FINITE_DIFF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/functor/apply_scalar_binary.hpp>
+#include <stan/math/prim/functor/finite_diff_gradient_auto.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <stan/math/prim/fun/sum.hpp>
+#include <stan/math/prim/fun/serializer.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+/**
+ * Helper function for aggregating tangents if the respective input argument
+ * was an fvar<T> type.
+ *
+ * Overload for when the input is not an fvar<T> and no tangents are needed.
+ *
+ * @tparam FuncTangent Type of tangent calculated by finite-differences
+ * @tparam InputArg Type of the function input argument
+ * @param tangent Calculated tangent
+ * @param arg Input argument
+ */
+template <typename FuncTangent, typename InputArg,
+          require_not_st_fvar<InputArg>* = nullptr>
+inline constexpr double aggregate_tangent(const FuncTangent& tangent,
+                                          const InputArg& arg) {
+  return 0;
+}
+
+/**
+ * Helper function for aggregating tangents if the respective input argument
+ * was an fvar<T> type.
+ *
+ * Overload for when the input is an fvar<T> and its tangent needs to be
+ * aggregated.
+ *
+ * @tparam FuncTangent Type of tangent calculated by finite-differences
+ * @tparam InputArg Type of the function input argument
+ * @param tangent Calculated tangent
+ * @param arg Input argument
+ */
+template <typename FuncTangent, typename InputArg,
+          require_st_fvar<InputArg>* = nullptr>
+auto aggregate_tangent(const FuncTangent& tangent, const InputArg& arg) {
+  return sum(apply_scalar_binary(
+      tangent, arg, [](const auto& x, const auto& y) { return x * y.d_; }));
+}
+}  // namespace internal
+
+/**
+ * This frameworks adds fvar<T> support for arbitrary functions through
+ * finite-differencing. Higher-order inputs (i.e., fvar<var> & fvar<fvar<T>>)
+ * are also implicitly supported.
```

Comment:
I don't like "this framework" I think it would be nice to talk about what the function does like 

```
Performs finite-differencing where the scalar type to be evaluated is `fvar<T>`
```